client/my-sites/menus/menu-item-types.js

```diff
@@ -171,23 +171,23 @@ MenuItemTypes.prototype.parse = function() {
 	this.fetched = true;
 
 	newTypes = types.filter( function( type ) {
-		return find( this._defaultItemTypes, { name: type.name } ) === undefined &&
-				type.api_queryable === true &&
-				type.map_meta_cap === true;
+		// With WP-API, post types self register via the `show_in_rest` flag, which replaces `api_queryable` from wpcom api
+		return ( find( this._defaultItemTypes, { name: type.slug } ) === undefined ) &&
+				( -1 === findIndex( this._excludedItemTypes, type.slug ) );
 	}, this );
 
 	debug( 'Found some new types', newTypes );
 
 	newTypes.forEach( function( type ) {
 		this.data.push( {
-			name: type.name,
+			name: type.slug,
 			family: 'post_type',
 			icon: 'standard',
 			renderer: 'renderPostOptions',
 			show: true,
-			label: type.label, //FIXME: how do we handle i18n here?
-			createLink: this.site.options.admin_url + 'post-new.php?post_type=' + type.name,
-			gaEventLabel: type.label
+			label: type.name, //FIXME: how do we handle i18n here?
```

Comment:
Good point.  I tested this out a bit and it looks like we'll need to implement something like our v1 `?locale=` parameter (p3hLNG-u8-p2).